### PR TITLE
SergeyT: add additional fixed formatting items

### PR DESCRIPTION
**Pull request by sergeyt**
rebasing PR #409

TODO:
* [ ] fix build
* [ ] consider fixing backward compatibility proposed in PR #409 

---
This change is [Reviewable](https://reviewable.io/reviews/chronotope/chrono/455)

### CODE REVIEW

File: src/format/mod.rs [hunk 1]
```diff
@@ -638,6 +672,41 @@ fn format_inner<'a>(
...
+                    time.map(|t| {
+                        result.push_str(if t.hour12().0 {"p"} else {"a"});
+                        Ok(())
+                    }),
+                &OneLetterUpperAmPm =>
+                    time.map(|t| {
+                        result.push_str(if t.hour12().0 {"P"} else {"A"});
+                        Ok(())
+                    }),
+                _ => return Err(fmt::Error),
```
Comment (quodlibetor) [inline src/format/mod.rs:707]: Could you explicitly match on `__DoNotMatch`? We're doing this internally, and are guaranteed to update in lockstep, we just want downstream users to know that they shouldn't.

File: src/format/mod.rs [hunk 2]
```diff
@@ -75,6 +75,10 @@ pub enum Numeric {
     /// Full Gregorian year (FW=4, PW=∞).
     /// May accept years before 1 BCE or after 9999 CE, given an initial sign.
     Year,
+    /// Full Gregorian year (FW=2, PW=∞). Can be negative.
+    ///
+    /// Prints last two digits for year >= 1000 or year <= -1000.
```
Comment (quodlibetor) [inline src/format/mod.rs:80]: Could you explicitly mention how ambiguous this is? E.g. that the years `1120` and `2020` will print the same year? And that it errors in years outside of the mentioned range?

Also, why does this error for N `-1000 < N < 1000`?

File: src/format/parsed.rs [hunk 3]
```diff
@@ -166,6 +166,14 @@ impl Parsed {
         set_if_consistent(&mut self.year, value.to_i32().ok_or(OUT_OF_RANGE)?)
     }
 
+    /// Tries to set the [`year`](#structfield.year) field from given value.
+    #[inline]
+    pub fn set_year_short(&mut self, value: i64) -> ParseResult<()> {
```
Comment (quodlibetor) [inline src/format/parsed.rs:171]: Is this used? I don't see it anywhere else in this PR. (possibly my search is broken)

File: src/format/mod.rs [hunk 4]
```diff
@@ -268,6 +297,8 @@ pub enum Item<'a> {
     Numeric(Numeric, Pad),
     /// Fixed-format item.
     Fixed(Fixed),
+    /// Extra fixed-format item.
+    FixedExt(FixedExt),
```
Comment (quodlibetor) [inline src/format/mod.rs:301]: could you remove this from `Item`

**Comment by sergeyt** [comment]
@quodlibetor i'll try to fix this PR soon. thanks for detailed review notes attached to the code, it is timesaver

**Comment by sergeyt** [comment]
@quodlibetor I am not sure I understand how to integrated these `Fixed` formats if I remove `FixedExt` from `Item`. Should I extend `InternalFixed`?

**Comment by quodlibetor** [comment]
> I am not sure I understand how to integrated these Fixed formats if I remove FixedExt from Item. Should I extend InternalFixed?

Exactly, yes.

File: src/format/mod.rs (hunk 4 above)
Comment (pitdicker) [reply src/format/mod.rs:301]: Agreed. Adding `FixedExt` would be a breaking change. We (now) have `FixedInternal` to but new variants in.

**Review by pitdicker — COMMENTED** [review]
It has been three years, sorry for the hiatus!

@sergeyt are you still interested in continuing this work?

To give my personal opinion: I don't think single-letter month and day names are useful enough to add to chrono. They are ambiguous, uncommon, and can't be used for parsing.

**Comment by sergeyt** [comment]
> It has been three years, sorry for the hiatus!
> 
> @sergeyt are you still interested in continuing this work?
> 
> To give my personal opinion: I don't think single-letter month and day names are useful enough to add to chrono. They are ambiguous, uncommon, and can't be used for parsing.

no problem. it is not terribly important now

**Pull request closed without merging**